src/domain/settings/regions/edit/general-section/index.tsx

```diff
@@ -0,0 +1,207 @@
+    })
+
+    if (shouldDelete) {
+      mutate(undefined, {
+        onSuccess: () => {
+          navigate("/a/settings/regions", {
+            replace: true,
+          })
+          notification("Success", "Region has been deleted", "success")
+          navigate(`/a/settings/regions`, { replace: true })
+        },
+        onError: (error) => {
+          notification("Error", getErrorMessage(error), "error")
+        },
+      })
+    }
+  }
+
+  return (
+    <>
+      <Section
+        title={region.name}
+        actions={[
+          {
+            label: "Edit Region Details",
+            onClick: toggle,
+            icon: <EditIcon size={20} className="text-grey-50" />,
+          },
+          {
+            label: "Delete Region",
+            onClick: handleDelete,
+            icon: <TrashIcon size={20} />,
+            variant: "danger",
+          },
+        ]}
+      >
+        <div className="flex flex-col gap-y-xsmall mt-large">
+          <h2 className="inter-large-semibold">Details</h2>
+          <div className="flex flex-col gap-y-xsmall">
+            <RegionDetail title={"Currency"}>
+              <div className="flex items-center gap-x-xsmall">
+                <span className="inter-base-semibold text-grey-90">
+                  {region.currency_code.toUpperCase()}
+                </span>
+                <span>
+                  {currencies[region.currency_code.toUpperCase()].name}
+                </span>
+              </div>
+            </RegionDetail>
+            <RegionDetail title={"Countries"}>
+              <div>
+                {region.countries && region.countries.length ? (
+                  <div className="flex items-center gap-x-xsmall">
+                    <p>
+                      {region.countries
+                        .slice(0, 4)
+                        .map((c) => c.display_name)
+                        .join(", ")}
+                    </p>
+                    {region.countries.length > 4 && (
+                      <Tooltip
+                        side="right"
+                        content={
+                          <ul>
+                            {region.countries.slice(4).map((c) => {
+                              return <li key={c.id}>{c.display_name}</li>
+                            })}
+                          </ul>
+                        }
+                      >
+                        <span className="cursor-default">
+                          + {region.countries.length - 4} more
+                        </span>
+                      </Tooltip>
+                    )}
+                  </div>
+                ) : (
+                  <p>No countries configured</p>
+                )}
+              </div>
+            </RegionDetail>
+            <RegionDetail title={"Payment providers"}>
+              <div>
+                {region.payment_providers && region.payment_providers.length ? (
```

Comment:
suggestion: Should we have an util for that behaviour, something like
```ts
const method = ({
  collection,
  mapperFunction,
  labelProp,
  itemCountToDisplay,
}) => {
  return (
    <>
      {collection?.length ? (
        <div className="flex items-center gap-x-xsmall">
          <p>
            {collection.payment_providers
              .slice(0, itemCountToDisplay)
              .map((p) => mapperFunction(p.id)[labelProp])
              .join(", ")}
          </p>
          {collection.length > 4 && (
            <Tooltip
              side="right"
              content={
                <ul>
                  {collection.slice(itemCountToDisplay).map((p) => {
                    return <li key={p.id}>{mapperFunction(p.id)}</li>
                  })}
                </ul>
              }
            >
              <span className="cursor-default">
                + {collection.length - 4} more
              </span>
            </Tooltip>
          )}
        </div>
      ) : (
        <p>No payment providers configured</p>
      )}
    </>
  )
}
```

and then we can use it like:

```ts
{
    method({
      collection: region.payment_providers,
      mapperFunction: paymentProvidersMapper,
      labelProp: "label",
      itemCountToDisplay: 4
    })
}
```

Of course I did not typed it and the mapper function can be optional and the style need to be managed :)
what do you think? since that behaviour is used multiple times I think it could be nice to centralise and make it generic